docs/design/service_traffic_policy.md

```diff
@@ -315,3 +297,128 @@ where packet leaves the node and goes back to the external entity that initiated
+
+We have a service of type `NodePort` where `internalTrafficPolicy: Local`:
+
+```
+$ oc get svc
+NAMESPACE        NAME            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                  AGE
+default          hello-world-2   NodePort    10.96.61.132   172.18.0.9    80:31358/TCP             108m
+$ oc get ep
+NAME            ENDPOINTS                         AGE
+hello-world-2   10.244.0.6:8080,10.244.1.3:8080   111m
+```
+
+This service is backed by two ovn pods:
+```
+$ oc get pods -owide -n surya
+NAME                            READY   STATUS    RESTARTS   AGE    IP           NODE          NOMINATED NODE   READINESS GATES
+hello-world-2-5c87676b7-h6rm5   1/1     Running   0          111m   10.244.0.6   ovn-worker    <none>           <none>
+hello-world-2-5c87676b7-l82w8   1/1     Running   0          111m   10.244.1.3   ovn-worker2   <none>           <none>
+```
+
+Traffic from pod hits the load balancer on the switch where the non-local endpoints for clusterIPs are filtered out. So traffic will be DNAT-ed to local endpoints if any. Here is how the load balancers will look like on the three KIND worker nodes for the above service.
+
+```
+a7865d9f-43d2-4cf9-a316-fc35e8c357a8    Service_surya/he    tcp        10.96.61.132:80        
+                                                            tcp        172.18.0.4:31358       10.244.0.6:8080,10.244.1.3:8080
+                                                            tcp        172.18.0.9:80          10.244.0.6:8080,10.244.1.3:8080
+d58c012a-7b1f-45ad-a817-a86845fde164    Service_surya/he    tcp        10.96.61.132:80        10.244.0.6:8080
+                                                            tcp        172.18.0.2:31358       10.244.0.6:8080,10.244.1.3:8080
+                                                            tcp        172.18.0.9:80          10.244.0.6:8080,10.244.1.3:8080
+a993003a-a177-4673-8f7c-825e2c9bf205    Service_surya/he    tcp        10.96.61.132:80        10.244.1.3:8080
+                                                            tcp        172.18.0.3:31358       10.244.0.6:8080,10.244.1.3:8080
+                                                            tcp        172.18.0.9:80          10.244.0.6:8080,10.244.1.3:8080
+```
+
+Once the packet is DNAT-ed it is then redirected to the backend pod.
+
+### **Host Traffic**
+
+This section will cover the networking entities hit when traffic travels from a cluster host via a clusterIP service to either host
+networked pods or cluster networked pods. Note that host to clusterIP is considered "internal" traffic.
+
+### **Host -> Service (ClusterIP) -> OVN Pod**
+
+1. Packet generated from the host towards clusterIP service `10.96.61.132:80` is marked for forwarding in the `mangle` table by an IP table rule called from the `OUTPUT` chain:
+
+```
+-A OUTPUT -j OVN-KUBE-ITP
+[1:60] -A OVN-KUBE-ITP -d 10.96.61.132/32 -p tcp -m tcp --dport 80 -j MARK --set-xmark 0x1745ec/0xffffffff
```

Comment:
`-m` u didn't use it in the code below so lets remove it to match the implementation

Reply:
so that's an IPTables magic, as soon as I add a `-p tcp` it knows its supposed to match on protocol and adds the `-m tcp`. I literally got the rule from my KIND cluster for this exact same code. So
```
{
			table: "mangle",
			chain: iptableITPChain,
			args: []string{
				"-p", string(svcPort.Protocol),
				"-d", string(clusterIP),
				"--dport", fmt.Sprintf("%d", svcPort.Port),
				"-j", "MARK",
				"--set-xmark", string(ovnkubeITPMark),
			},
			protocol: protocol,
		},
```
translates to:
`-A OVN-KUBE-ITP -d 10.96.197.155/32 -p tcp -m tcp --dport 80 -j MARK --set-xmark 0x1745ec/0xffffffff`

Reply:
Indeed, it's safe to assume that the `tcp` module has been loaded by this point :-)